File: data/test_cases/idor/secure_idor.js
// Secure: verifies the caller owns the resource, satisfies BAC-001 / BAC-002
const express = require('express');
const router = express.Router();

router.get('/calendar/:userId', requireAuth, async (req, res) => {
  if (req.params.userId !== req.user.id) {
    return res.status(403).json({ error: 'forbidden' });
  }
  const calendar = await getCalendar(req.params.userId);
  res.json(calendar);
});

module.exports = router;

File: data/test_cases/idor/secure_idor.py
# Secure: verifies the caller owns the resource, satisfies BAC-001
@app.get("/calendar/{user_id}")
async def calendar(user_id: int, current_user=Depends(get_current_user)):
    if user_id != current_user.id:
        raise HTTPException(status_code=403)
    return get_calendar(user_id)
